### ISSUE TITLE
Dashboard panics on getting heapster selector when stateful sets not supported by cluster

### ISSUE THREAD

**Issue by MerlinPong** [kubernetes/dashboard#1916]
##### Environment

```
Dashboard version:
Kubernetes version:
Operating system:
Node.js version:
Go version:
```
##### Observed result
Dashboard reported Unknown Server Error (500):
```
[restful] recover from panic situation: - Failed to create heapster selector for resource "statefulset". Error: GetMyPodsFromCache: pods were not available in cache. Required for resource type: "statefulset"
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/runtime/asm_amd64.s:472
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/runtime/panic.go:426
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/dataselect/dataselect.go:157
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/dataselect/dataselect.go:210
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/statefulset/statefulsetlist/statefulsetlist.go:116
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/statefulset/statefulsetlist/statefulsetlist.go:100
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/resource/statefulset/statefulsetlist/statefulsetlist.go:68
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/apihandler.go:622
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/apihandler.go:523
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:267
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:21
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/apihandler.go:135
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:19
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/apihandler.go:97
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:19
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/src/app/backend/handler/apihandler.go:153
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/filter.go:19
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:269
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:120
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/net/http/server.go:1618
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/net/http/server.go:1910
    /home/travis/build/kubernetes/dashboard/.tmp/backend/src/github.com/kubernetes/dashboard/vendor/github.com/emicklei/go-restful/container.go:287
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/net/http/server.go:1910
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/net/http/server.go:2081
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/net/http/server.go:1472
    /home/travis/.gimme/versions/go1.6.1.linux.amd64/src/runtime/asm_amd64.s:1998

```

**Comment by floreks** [comment]
What is your kubernetes version? I have found only 1 situation where this error might occur and it is if cluster does not support stateful sets. We should fix that anyway.

**Comment by aliakhtar** [comment]
I'm encountering this issue as well:

```
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.4", GitCommit:"d6f433224538d4f9ca2f7ae19b252e6fcb66a3ae", GitTreeState:"clean", BuildDate:"2017-05-19T18:44:27Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.2", GitCommit:"477efc3cbe6a7effca06bd1452fa356e2201e1ee", GitTreeState:"clean", BuildDate:"2017-04-19T20:22:08Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}

```

My cluster only had one ss and pod when this happened - deleting and re-creating the ss caused the error to go away. Will post stacktrace if i encounter it again

**Comment by floreks** [comment]
We will remove this check and return no metrics instead of panic.

**Comment by floreks** [comment]
I have checked and this is already fixed by #2017.

**Comment by jasonyihk** [comment]
one side note: the same error will pop up if you have typo in the yaml file for statefulset. e.g. imagePullPolicy: **always**, which should be **Always**. in case anyone run into the same issue.
